core/services/assets/Registry.php

```diff
@@ -135,61 +102,99 @@ public function getI18nRegistry()
-     * Callback for the WP script actions.
-     * Used to register globally accessible core scripts.
-     * Also used to add the eejs.data object to the source for any js having eejs-core as a dependency.
+     * Callback for the wp_enqueue_scripts actions used to register assets.
      *
+     * @since $VID:$
+     * @throws AssetRegistrationException
+     * @throws InvalidDataTypeException
+     * @throws Exception
      */
-    public function scripts()
+    public function registerScriptsAndStyles()
     {
-        global $wp_version;
-        wp_register_script(
-            'ee-manifest',
-            $this->getJsUrl(self::ASSET_NAMESPACE, 'manifest'),
-            array(),
-            null,
-            true
-        );
-        wp_register_script(
-            'eejs-core',
-            $this->getJsUrl(self::ASSET_NAMESPACE, 'eejs'),
-            array('ee-manifest'),
-            null,
-            true
-        );
-        wp_register_script(
-            'ee-vendor-react',
-            $this->getJsUrl(self::ASSET_NAMESPACE, 'reactVendor'),
-            array('eejs-core'),
-            null,
-            true
-        );
-        //only run this if WordPress 4.4.0 > is in use.
-        if (version_compare($wp_version, '4.4.0', '>')) {
-            //js.api
-            wp_register_script(
-                'eejs-api',
-                EE_LIBRARIES_URL . 'rest_api/assets/js/eejs-api.min.js',
-                array('underscore', 'eejs-core'),
-                EVENT_ESPRESSO_VERSION,
-                true
-            );
-            $this->jsdata['eejs_api_nonce'] = wp_create_nonce('wp_rest');
-            $this->jsdata['paths'] = array('rest_route' => rest_url('ee/v4.8.36/'));
+        try {
+            $this->registerScripts($this->assets->getJavascriptAssets());
+            $this->registerStyles($this->assets->getStylesheetAssets());
+        } catch (Exception $exception) {
+            new ExceptionStackTraceDisplay($exception);
         }
-        if (! is_admin()) {
-            $this->loadCoreCss();
+    }
+
+
+    /**
+     * Registers JS assets with WP core
+     *
+     * @since $VID:$
+     * @param JavascriptAsset[] $scripts
+     * @throws AssetRegistrationException
+     * @throws InvalidDataTypeException
+     */
+    public function registerScripts(array $scripts)
+    {
+        foreach ($scripts as $script) {
+            // skip to next script if this has already been done
+            if($script->isRegistered()) {
+                continue;
+            }
+            do_action(
+                'AHEE__EventEspresso_core_services_assets_Registry__registerScripts__before_script',
+                $script
+            );
+            $registered = wp_register_script(
+                $script->handle(),
+                $script->source(),
+                $script->dependencies(),
+                $script->version(),
+                $script->loadInFooter()
+            );
+            if(WP_DEBUG && ! $registered) {
```

Comment:
I know in the past @joshfeck has expressed reluctance for us throwing exceptions even when `WP_DEBUG` is true.  I'm on the fence here but wonder if it should be run by him first?  I still would like to know what conditions would cause an asset to fail getting registered.  If its when there's already a script handle registered, then that is probably less likely to occur the way we're doing our registrations, which means this condition for throwing an exception is probably fine.

Reply:
`$registered` might not be `true` if the asset is already registered or if some other issue occurs that prevents things from working (ex: there's a `doing_it_wrong()` notice that could be triggered but I can't remember if those bring the card house down or not) and you are correct that if "there's already a script handle registered, then that is probably less likely to occur the way we're doing our registrations".

Reply:
Could you check for `EE_DEBUG` instead?

Reply:
sure, will do